app/models/task.rb

```diff
@@ -76,13 +78,13 @@ def remove_watcher(user)
   end
 
   def assign!(assignee_id, by_user)
-    result = update(assignee_id: assignee_id)
+    result = update(assignee_id: assignee_id, assigned_by: by_user.id)
     send_notifications("Task '#{self.title}' has been assigned to #{assignee.full_name} by #{by_user.full_name}", self.watchers - [by_user]) if result
     result
   end
 
   def complete!(by_user)
-    self.update(completed_at: Time.now)
+    self.update(completed_at: Time.now, completed_by: by_user.id)
```

Comment:
add "_id" to column name. It's more readable
and chance code to `...completed_by: user)`